### PR TITLE
radosgw: Make compilation with CryptoPP possible.

### PR DESCRIPTION
**Pull request by aclamk**
Signed-off-by: Adam Kupczyk <akupczyk@mirantis.com>

### CODE REVIEW

File: src/rgw/rgw_crypt.cc [hunk 1]
```diff
@@ -24,6 +24,19 @@
...
 #include <cryptopp/modes.h>
 #include <cryptopp/aes.h>
+
+/*
+ * patching for https://github.com/openssl/openssl/issues/1437
+ * newer versions of civetweb do not experience problems
+ */
+extern "C"
+{
+bool SSL_CTX_set_ecdh_auto(void* dummy, int onoff)
```
Comment (tchaikov) [inline src/rgw/rgw_crypt.cc]: thanks, this addresses the issue mentioned at https://github.com/ceph/ceph/pull/13315#issuecomment-283617562, but shall we just update the civetweb submodule?

Comment (aclamk) [inline src/rgw/rgw_crypt.cc]: @yehudasa I do not feel comfortable answering question about updating civetweb. Do we know if we can safely update civetweb no newer version?
@tchaikov I was not aware that this problem affects more then radosgw. In that case this code can be moved to some generic ceph parts like ceph_crypto.cc

Comment (tchaikov) [inline src/rgw/rgw_crypt.cc]: @aclamk i think it's just rgw.

**Comment by liewegas** [comment]
@yehudasa ?

**Comment by aclamk** [comment]
At sync-up it was decided that we will not change civetweb for now. So, presented solution remains.

File: src/rgw/rgw_crypt.cc [hunk 2]
```diff
@@ -24,6 +24,24 @@
...
+#else
+/*
+ * patching for https://github.com/openssl/openssl/issues/1437
+ * newer versions of civetweb do not experience problems
+ */
+extern "C"
+{
+bool SSL_CTX_set_ecdh_auto(void* dummy, int onoff)
+{
+  return onoff!=0;
```
Comment (tchaikov) [inline src/rgw/rgw_crypt.cc]: please add a space around `!=`.

Comment (aclamk) [inline src/rgw/rgw_crypt.cc]: @tchaikov done.

File: src/rgw/rgw_crypt.cc [hunk 3]
```diff
@@ -24,6 +24,24 @@
 #include <cryptopp/cryptlib.h>
 #include <cryptopp/modes.h>
 #include <cryptopp/aes.h>
+#if defined(CIVETWEB_VERSION_MAJOR) && defined(CIVETWEB_VERSION_MINOR) \
+  && ((CIVETWEB_VERSION_MAJOR>=1 && CIVETWEB_VERSION_MINOR>=10) || \
+      (CIVETWEB_VERSION_MAJOR>1))
```
Comment (tchaikov) [inline src/rgw/rgw_crypt.cc]: i think this line should be 
```c
   (CIVETWEB_VERSION_MAJOR>1))
```

but maybe following macro is easier to digest?
```c
#if defined(CIVETWEB_VERSION_MAJOR) && \
    defined(CIVETWEB_VERSION_MINOR) &&
    (CIVETWEB_VERSION_MAJOR * 100) + CIVETWEB_VERSION_MINOR >= 110
```

Comment (aclamk) [reply src/rgw/rgw_crypt.cc]: @tchaikov Basically I like your solution better.

**Comment by tchaikov** [comment]
@aclamk could you fold these two commits into a single one, then i think it'd be good to merge as long as shaman is happy.

File: src/rgw/rgw_crypt.cc (hunk 2 above)
Comment (tchaikov) [reply src/rgw/rgw_crypt.cc]: @aclamk this change is reverted somehow?

**Pull request merged**